### DISCUSSION TITLE
Posibility to limit the power as in WLED for the AWA protocol

### DISCUSSION THREAD

**Discussion by Bascht74** [awawa-dev/HyperHDR#33] [Q&A]
Hi @awawa-dev , 

I am new HyperHDR and looking for an possibility to limit the power usage of the LEDs controlled by HyperHDR / HyperSerialWLED / HyperSerialEsp8266. I have 360 LEDs (86" TV) that need more than 20A@5V and I want to limit that to max. 20A. 
I know there are settings in WLED, but I don't know if these settings are used within HyperSerialWLED in AWA-Mode or HyperSerialEsp8266 with AWA.
... and I din't find any setting within HyperHDR to limit the brightness dynamically (not for one LED but brightness in total = total power limit), so that not more than 20A are used...

Do you have any information how to do that?

If it is not possible I have no choice and would buy 2x 15A power supplies like and power two corners with each seperatly: 
https://www.amazon.de/ALITOVE-Adapter-Converter-Charger-5-5x2-1mm/dp/B06XRS5F21/ref=sr_1_5?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&dchild=1&keywords=5V+20A+Netzteil&qid=1613390075&sr=8-5

Thanks for any advice about that and thanks for your great work on HyperHDR!!!! Almost no lag in my test setup!

Sebastian

**Comment by mattman73** [comment]
Hi Sebastian
Not sure if this helps but you could run a 30a PSU, https://www.amazon.co.uk/Universal-Regulated-Switching-Converter-Transformer-5V-30A-150W/dp/B07PRY73TN/ref=sr_1_4?crid=2BAX3P9E4TT6N&dchild=1&keywords=5v+30a+power+supply&qid=1613394540&sprefix=5v+30%2Caps%2C180&sr=8-4
Matt

**Comment by awawa-dev** [comment]
In HyperSerialWLED data are passed to WLED engine and they should be processed using all user settings in WLED. At least it should be done that way because otherwise the standard adalight@115200 interface in WLED would be broken as well and it would mean it's a bug in that app. Didn't try it but you could lower in WLED interface power limit to very low value and see if the leds are turn down while using HyperHDR.

**Reply by Bascht74** [reply]
@awawa-dev Thanks for the information. 
I don't want to use that kind of power supply because of the "holes" and the hight voltage parts... :-)
I will try it out with your WLED fork then...

On the other side your answer means that there ist no way in HyperSerialEsp8266 or HyperHDR itself to dynamicly limit the brightness (= power used), and you won't implement this, right?
Right now I am using a Pi4 with SK6812 directly (no issues so far), but I want to use your ESP8266 way because of the white calibration, that is not possible in hyperHDR itself. Or am I wrong?

About the ESP8266 part: 
I read your blog entry here: 
https://hyperhdr.blogspot.com/2020/12/my-build-log-using-sk6812-rgbw-led.html
and ordered all the parts, except the case.
Could you add an link for that specific case, you used in your blog post?

I saw the pictures of your implementation for Esp8266 Wemos d1 mini and some look slightly different on the pictures.
Are these two variantions of it?

Thanks again for your advice in that.

**Reply by awawa-dev** [reply]
Right now I don't plan to implemented power limit because I'm quite busy implementing a lot of features that will come in hyperhdr v15.

There is possibility that 'the power limit' will come to HyperSerialEsp8266 after the release of HyperHDR. But it won't be called 'power limit' because it's misleading: as I measured my RGBW after calibration it doesn't even come close to 60% of calculated power usage. Maybe because of not all white range is used. So it will be rather 'brightness limiter with scaling' or something.

The case for my unit is a product manufactured in Poland  but it can be purchased on ebay also: https://www.ebay.co.uk/itm/Z17W-Enclosure-with-panel-vented-X-218mm-Y-237mm-Z-92mm-black-KRADEX-/193703339919

For Esp8266 wemos d1 mini: there are a lot of different clones...even they sometimes sold ESP32 as wemos d1 mini.
It must contains Esp8266 and ch340g, similar pinout to mine is also a nice thing because you don't have to experiment then. Check seller's feedback because theirs quality can varies. 

If you plan to build led strip from scratch then I strongly recommend to rotate sk6812 by 45 angle. I used something like that attached to the frame in my new setup: https://images.obi.pl/product/PL/1065x357/309081_1.jpg

**Reply by Bascht74** [reply]
Thanks for the infos. 
I soldered my ESP8266 today and it is working great!
I got now 2 x 10A power supplies and connected that to 2 corners each. So that is now working with 300 LEDs...

I will look for that case, thanks for the link!

Thanks again for the explanation of the CH340G, I bought ESP8266 with it and it is working good so far...
Now an P4 & ESP8266 (so far without extra LED / level shifter because no LED left) is doing fine...

Looking forward to the next version...
